air/src/execution_step/air/ap/apply_to_arguments.rs

```diff
@@ -118,3 +119,49 @@ fn apply_scalar_wl_impl(
 
     Ok(result)
 }
+
+fn apply_canon_stream(
+    canon_stream: &ast::CanonStreamWithLambda<'_>,
+    exec_ctx: &ExecutionCtx<'_>,
+    trace_ctx: &TraceHandler,
+) -> ExecutionResult<ValueAggregate> {
+    match &canon_stream.lambda {
+        Some(lambda) => apply_canon_stream_with_lambda(canon_stream.name, lambda, exec_ctx, trace_ctx),
+        None => apply_canon_stream_without_lambda(canon_stream.name, exec_ctx, trace_ctx),
+    }
+}
+
+fn apply_canon_stream_with_lambda(
+    stream_name: &str,
+    lambda: &LambdaAST<'_>,
+    exec_ctx: &ExecutionCtx<'_>,
+    trace_ctx: &TraceHandler,
+) -> ExecutionResult<ValueAggregate> {
+    use crate::execution_step::boxed_value::JValuable;
+
+    let canon_stream = exec_ctx.scalars.get_canon_stream(stream_name)?;
+    let jvaluable = &canon_stream as &dyn JValuable;
```

Comment:
We need some form of a cast here to use `as_jvalue` method of the `JValuable` trait. It's almost free, it creates a fat pointer on a stack which consist of pointer to a `vtable` and data. And I think that it could be optimized by a compiler.

Reply:
One should use fully qualified trait call `JValuable::as_value(...)` for that.